experiment/measurer/measure_worker.py

```diff
@@ -95,17 +98,18 @@ def _get_previous_cycle_state(self) -> list:
 
         return json.loads(result.output)
 
-    def get_previous(self):
+    def get_previous(self, cycle_dependent=True):
         """Returns the previous state."""
         if self._prev_state is None:
-            self._prev_state = self._get_previous_cycle_state()
+            self._prev_state = self._get_previous_cycle_state(
+                cycle_dependent=cycle_dependent)
 
         return self._prev_state
 
-    def set_current(self, state):
+    def set_current(self, state, overwrite=False):
```

Comment:
Also use cycle_dependent instead. Otherwise, there are two concepts where there should be one.